lib/src/cpp/common/LocatorList.hpp

```diff
@@ -0,0 +1,367 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ */
+
+#ifndef _FAST_DDS_QOS_PROFILES_MANAGER_COMMON_LOCATOR_LIST_HPP_
+#define _FAST_DDS_QOS_PROFILES_MANAGER_COMMON_LOCATOR_LIST_HPP_
+
+#include <string>
+
+#include <xercesc/dom/DOM.hpp>
+
+namespace eprosima {
+namespace qosprof {
+namespace common {
+namespace locator_list {
+
+/************************************************************************/
+/* Print functions                                                      */
+/************************************************************************/
+
+/**
+ * @brief Parse XML node and print the locator.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed. If empty, the complete list is printed.
+ *
+ * @return std::string XML section containing the locator.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator kind.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator kind.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
```

Comment:
```suggestion
 * @throw BadParameter Exception if the index is empty or not an integer.
```

This method does not allow an empty index. This is checked and ensured in the CLI/GUI, but it should be also checked within the function for safety reasons. Please, apply to the other similar functions.